src/test/optimizer/optimizer_test.cpp

```diff
@@ -142,7 +228,8 @@ TEST_F(OptimizerTest, OptimizesSubqueriesExactlyOnce) {
   Optimizer optimizer{};
   optimizer.add_rule(std::move(rule));
 
-  optimizer.optimize(lqp);
+  const auto optimized_lqp = optimizer.optimize(std::move(lqp));
+  lqp = nullptr;
```

Comment:
Why do we specifically set `lqp = nullptr` here and don't do that above?

Reply:
Mmh, no real reason other than that there were more than a couple of lines below. In the end, both should work, so I would just leave both versions.